WordPress/Classes/ViewRelated/Post/PostListViewController.swift

```diff
@@ -0,0 +1,685 @@
+        
+        // If we have recently trashed posts, create an OR predicate to find posts matching the filter,
+        // or posts that were recently deleted.
+        if let recentlyTrashedPostObjectIDs = recentlyTrashedPostObjectIDs
+            where searchText?.characters.count == 0 && recentlyTrashedPostObjectIDs.count > 0 {
+            
+            let trashedPredicate = NSPredicate(format: "SELF IN %@", recentlyTrashedPostObjectIDs)
+            
+            if let originalFilterPredicate = filterPredicate {
+                filterPredicate = NSCompoundPredicate(orPredicateWithSubpredicates: [originalFilterPredicate, trashedPredicate])
+            } else {
+                filterPredicate = trashedPredicate
+            }
+        }
+        
+        if let filterPredicate = filterPredicate {
+            predicates.append(filterPredicate)
+        }
+        
+        if shouldShowOnlyMyPosts() {
+            let myAuthorID = blog?.account.userID ?? 0
+            
+            // Brand new local drafts have an authorID of 0.
+            let authorPredicate = NSPredicate(format: "authorID = %@ || authorID = 0", myAuthorID)
+            predicates.append(authorPredicate)
+        }
+        
+        if let searchText = searchText where searchText.characters.count > 0 {
+            let searchPredicate = NSPredicate(format: "postTitle CONTAINS[cd] %@", searchText)
+            predicates.append(searchPredicate)
+        }
+        
+        let predicate = NSCompoundPredicate(andPredicateWithSubpredicates: predicates)
+        return predicate
+    }
+    
+    // MARK: - Table View Handling
+    
+    private func tableView(tableView: UITableView, estimatedHeightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        let post = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as! Post
+        
+        if cellIdentifierForPost(post) == self.dynamicType.postCardRestoreCellIdentifier {
+            return CGFloat(self.dynamicType.postCardRestoreCellRowHeight)
+        }
+        
+        return CGFloat(self.dynamicType.postCardEstimatedRowHeight)
+    }
+    
+    private func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        let width = CGRectGetWidth(tableView.bounds)
+        return self.tableView(tableView, heightForRowAtIndexPath: indexPath, forWidth: width)
+    }
+   
+    private func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath, forWidth width: CGFloat) -> CGFloat {
+        guard let post = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as? Post else {
+            return 0
+        }
+        
+        if cellIdentifierForPost(post) == self.dynamicType.postCardRestoreCellIdentifier {
+            return CGFloat(self.dynamicType.postCardRestoreCellRowHeight)
+        }
+        
+        var cell : PostCardTableViewCell!
```

Comment:
Since `cell` is initially `nil`, it needs to either be `?` or `!`.

In this case, the variable is `nil` when declared, but set to a value immediately after that, which is the [suggested used case](https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/TheBasics.html#//apple_ref/doc/uid/TP40014097-CH5-ID309) for IUOs:

```Implicitly unwrapped optionals are useful when an optional’s value is confirmed to exist immediately after the optional is first defined and can definitely be assumed to exist at every point thereafter.```

A nicer option would be to create a method to return the correct cell, so that this doesn't need to be an IUO.  I didn't do this because I wanted to try to maintain parity for the sake of verifying the migration, but I can do it if you feel that's necessary.  The code above would become something like:

```swift
let cell = cellForPost(post)
```